source/geod24/LocalRest.d

```diff
@@ -553,31 +711,78 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
             // Control the node behavior
             Control control;
 
+            struct AwaitingMessages
+            {
+                /// Message
+                Variant var;
+                /// Originating `Connection`
+                Connection conn;
+            }
             // we need to keep track of messages which were ignored when
             // node.sleep() was used, and then handle each message in sequence.
-            Variant[] await_msgs;
+            AwaitingMessages[] await_msgs;
 
             scheduler.start(() {
-                while (1)
+                while (true)
                 {
+                    Connection new_conn;
+                    Response res;
                     Variant msg;
-                    if (chn.read(msg, 10.msecs))
+
+                    C.SelectEntry[] read_list;
+                    C.SelectEntry[] write_list;
+
+                    foreach (ref conn; incoming_conns)
+                        if (!conn.command_chn.isClosed())
+                            read_list ~= C.SelectEntry(conn.command_chn, &msg);
+                    foreach (ref conn; outgoing_conns)
+                        if (!conn.resp_chn.isClosed())
+                            read_list ~= C.SelectEntry(conn.resp_chn, &res);
+                    read_list ~= C.SelectEntry(bind_chn, &new_conn);
+
+                    auto sel_ret = C.select(read_list, write_list, 10.msecs);
+
+                    if (!control.isSleeping())
+                    {
+                        foreach (ref await_msg; await_msgs)
+                            handleCommand(await_msg.var.cmd, node, control.filter, await_msg.conn.resp_chn);
+                        await_msgs.length = 0;
+                        assumeSafeAppend(await_msgs);
+                    }
+
+                    if (!sel_ret.success)
+                        continue;
+
+                    // Bind chn
+                    if (auto bind_chn = cast(BindChn) read_list[sel_ret.id].selectable)
```

Comment:
`bid_chn` doesn't seem to be used here. So how about changing it to `if (cast(BindChn) read_list[sel_ret.id].selectable)`?